ports/esp32/mphalport.h

```diff
@@ -136,6 +137,16 @@ static inline int mp_hal_pin_read(mp_hal_pin_obj_t pin) {
 static inline void mp_hal_pin_write(mp_hal_pin_obj_t pin, int v) {
     gpio_set_level(pin, v);
 }
+static inline int mp_hal_pin_output_level(mp_hal_pin_obj_t pin) {
+    #if defined(GPIO_OUT1_REG)
+    return (pin < 32
+        ? (*(uint32_t *)GPIO_OUT_REG & (1 << pin))
+        : (*(uint32_t *)GPIO_OUT1_REG & (1 << (pin - 32))))
+            ? 1 : 0;
+    #else
+    return (*(uint32_t *)GPIO_OUT_REG & (1 << pin)) ? 1 : 0;
```

Comment:
You should be able to do away with the `?:` operator by right shifting the register, eg (schematically):
```c
return (*GPIO_OUT_REG >> pin) & 1
```
(Same comment applies to the more complicated case when `GPIO_OUT1_REG` is defined.)

Reply:
Yes. That looks simpler. I wonder what the optimizer would have created with the previous code.